File: RajFABFrontend/src/services/api/appeal.ts
import { BaseApiService } from './base';

// Frontend Appeal interface matching DTO
export interface Appeal {
  id: string;
  factoryRegistrationNumber: string;
  dateOfAccident: string | null;
  dateOfInspection: string | null;
  noticeNumber: string | null;
  noticeDate: string | null;
  orderNumber: string | null;
  orderDate: string | null;
  factsAndGrounds: string | null;
  reliefSought: string | null;
  challanNumber: string | null;
  enclosureDetails1: string | null;
  enclosureDetails2: string | null;
  signatureOfOccupier: string | null;
  signature: string | null;
  place: string | null;
  date: string | null;
  version: number;
  isActive: boolean;
  isESignCompletedManager: boolean;
  isESignCompletedOccupier: boolean;
  createdAt: string;
  updatedAt: string;
  status: string;
  appealApplicationNumber: string;
  appealRegistrationNumber: string;
  applicationPDFUrl: string;
}
export interface AppealRes {
  html: string
}

// Payload for creating/updating an appeal
export interface CreateAppealRequest {
  factoryRegistrationNumber: string;
  dateOfAccident?: string;
  dateOfInspection?: string;
  noticeNumber?: string;
  noticeDate?: string;
  orderNumber?: string;
  orderDate?: string;
  factsAndGrounds?: string;
  reliefSought?: string;
  challanNumber?: string;
  enclosureDetails1?: string;
  enclosureDetails2?: string;
  signatureOfOccupier?: string;
  signature?: string;
  place?: string;
  date?: string;
}

export interface AppealRes {
  appealData: Appeal;
  estFullDetails: any
}
// Service class for API calls
export class AppealApiService extends BaseApiService {
  async getAll(): Promise<Appeal[]> {
    const response = await this.request<Appeal[] | { data: Appeal[] }>('/appeal/all');
    return Array.isArray(response) ? response : response.data ?? [];
  }

  async getById(id: string): Promise<AppealRes> {
    return this.request<AppealRes>(`/appeal/${id}`);
  }

  async create(appeal: CreateAppealRequest): Promise<AppealRes> {
    return this.request<AppealRes>('/appeal/create', {
      method: 'POST',
      body: JSON.stringify(appeal),
    });
  }

  async update(id: string, appeal: CreateAppealRequest): Promise<Appeal> {
    return this.request<Appeal>(`/appeal/${id}/update`, {
      method: 'POST',
      body: JSON.stringify(appeal),
    });
  }

  async delete(id: string): Promise<void> {
    await this.request<void>(`/appeal/${id}/delete`, {
      method: 'POST',
    });
  }
}

export const appealApi = new AppealApiService();

File: RajFABBackend/Models/FactoryModels/Appeal.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RajFabAPI.Models
{
    public class Appeal
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required]
        [StringLength(255)]
        public string FactoryRegistrationNumber { get; set; }
        
        [Required]
        [StringLength(255)]
        public string AppealRegistrationNumber { get; set; }

        [Required]
        [StringLength(255)]
        public string AppealApplicationNumber { get; set; } =  Guid.NewGuid().ToString().Replace("-","").Substring(0, 10).ToUpper();

        public DateTime? DateOfAccident { get; set; }
        public DateTime? DateOfInspection { get; set; }

        [StringLength(100)]
        public string NoticeNumber { get; set; }

        public DateTime? NoticeDate { get; set; }

        [StringLength(100)]
        public string OrderNumber { get; set; }

        public DateTime? OrderDate { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string FactsAndGrounds { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string ReliefSought { get; set; }

        [StringLength(100)]
        public string ChallanNumber { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string EnclosureDetails1 { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string EnclosureDetails2 { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string SignatureOfOccupier { get; set; }

        [Column(TypeName = "nvarchar(max)")]
        public string Signature { get; set; }

        [StringLength(100)]
        public string Place { get; set; }

        [StringLength(100)]
        public string Status { get; set; } = "Pending";

        public DateTime? Date { get; set; }

        [Column(TypeName = "decimal(3,1)")]
        public decimal Version { get; set; } = 1.0m;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;

        [StringLength(100)]
        public string? ESignPrnNumberOccupier { get; set; }

        [StringLength(100)]
        public string? ESignPrnNumberManager { get; set; }
        public bool IsESignCompletedOccupier { get; set; } = false;
        public bool IsESignCompletedManager { get; set; } = false;
        public string? ApplicationPDFUrl { get; set; }

        [StringLength(100)]
        public string? ESignPrnNumber { get; set; }
        public bool IsESignCompleted { get; set; } = false;
    }
}
